docs/NooBaaNonContainerized/S3Ops.md

```diff
@@ -40,6 +44,64 @@ The following lists describe the bucket and object operations available in NooBa
+In both cases the argument for the bucket policy is a string
+2. Removing a bucket policy: `noobaa-cli bucket update --name <bucket_name> --bucket_policy ''` (using empty string)
+3. Get a bucket policy: `noobaa-cli bucket status --name <bucket_name>` (a bucket policy will be printed in the bucket details)
+
+bucket_policy as a string example:
+`'{"Version":"2012-10-17","Statement":[{"Effect":"Allow","Principal":{"AWS":["<account-name>"]},"Action":["s3:*"],"Resource":["arn:aws:s3:::<bucket-name>/*","arn:aws:s3:::<bucket-name>"]}]}'`
+
+Replace `<account-name>` with account name and `<bucket-name>` with a bucket name.
+Note: `arn:aws:s3:::<bucket-name>` for S3 bucket operations and  `arn:aws:s3:::<bucket-name>/*` for S3 object operations.
+Warning: this policy allows `<account-name>` to run all S3 operations.
+
+#### Bucket Policy in S3 API (using AWS CLI)
+1. Adding bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api put-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+2. Removing a bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api delete-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+3. Get a bucket policy: `AWS_ACCESS_KEY_ID={access_key} AWS_SECRET_ACCESS_KEY={secret_key} aws s3api delete-bucket-policy --endpoint-url {endpoint_address} --bucket {bucket_name} --policy file://policy.json`
+
+policy.json example:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [ 
+    { 
+     "Effect": "Allow", 
+     "Principal": { "AWS": [ "<account-name>" ] }, 
+     "Action": [ "s3:*" ], 
+     "Resource": [ "arn:aws:s3:::<bucket-name>/*", "arn:aws:s3:::<bucket-name>" ] 
+    }
+  ]
+}
+```
+Replace `<account-name>` with account name and `<bucket-name>` with a bucket name.
+Note: `arn:aws:s3:::<bucket-name>` for S3 bucket operations and  `arn:aws:s3:::<bucket-name>/*` for S3 object operations.
+Warning: this policy allows `<account-name>` to run all S3 operations.
+
+##### Principal Field:
+A bucket policy defines which principals can perform actions on the bucket. The Principal element specifies the user or account that is either allowed or denied access to a resource.
+Currently we support a couple of options:
+1. Grant anonymous permissions (all principals): either `"Principal": { "AWS": "*" }` or `"Principal": { "*" }`.
+2. Principal by account name: `"Principal": { "AWS": [ "<account-name-1>", "<account-name-2>", ... ,"<account-name-n>"] }`
+
+##### Additional Supported Features:
+- "NotPrincipal", "NotAction" and "NotResource" policy elements in bucket policy (which can be used instead of the Principal, Action, and Resource).
+- Statements containing conditions for authenticating based on server side encryption algorithm - enabling deny, accept bucket policies for "stringEquals", "StringNotEquals" and "null" conditions for server side encryptions.
```

Comment:
@romayalon @nadavMiz please let me know if you want this part inside the docs.
I'm not familiar with this feature.

Reply:
Decide to remove this part at this point which is based on PRs:
Note: the part "Additional Supported Features" is based on the PRs description:
https://github.com/noobaa/noobaa-core/pull/7465
https://github.com/noobaa/noobaa-core/pull/7273